src/components/OracleQueryList/OracleQueryList.tsx

```diff
@@ -19,20 +18,19 @@ interface Props {
  * @param page The page the queries are being rendered on.
  * @param items The queries to render.
  * @param isLoading Whether the queries are still loading.
+ * @param findQueryIndex - the index of the query from the url, if any
  */
 export function OracleQueryList({
   page,
   items,
   isLoading,
   findQueryIndex,
 }: Props) {
-  const [itemsToShow, setItemsToShow] = useState(items);
+  const { entriesToShow, ...paginationProps } = usePagination(
```

Comment:
this new api is much cleaner and avoids duplicating the code which lead to the bug this pr is intended to fix